lib/GADS/Schema/ResultSet/Report.pm

```diff
@@ -0,0 +1,130 @@
+package GADS::Schema::ResultSet::Report;
+
+use strict;
+use warnings;
+
+use Data::Dumper;
+
+use Log::Report 'linkspace';
+
+use parent 'DBIx::Class::ResultSet';
+
+=head1 Package functions
+
+=head2 Load
+
+Function to load a report for a given id - it requires the report id, record id and the schema to be passed in and will return a report object
+
+=cut
+
+sub load {
+    my ( $self, $id, $record_id ) = @_;
+
+    my $schema = $self->result_source->schema;
+
+    error "Invalid report id provided"
+      unless $id && $id =~ /^\d+$/;
+
+    my $result = $self->find( { id => $id }, { prefetch => 'report_layouts' } )
+      or error "No report found for id $id";
+    $result->schema($schema) if !$result->schema;
```

Comment:
Not sure the above is needed?

Reply:
It is needed to give the linked schema to the Report object, otherwise it is undef.